### PR TITLE
Set HDF5_ROOT to avoid hardcoding library dependencies

### PR DESCRIPTION
**Pull request by paulromano**
Currently, the DAGMC package on conda forge has hardcoded libraries in the exported `DAGMCTargets.cmake` file from a temporary build path which won't show up on a user's system, which breaks downstream dependencies (e.g., OpenMC). After a lot of digging, I've determined that it is the fault of HDF5. See here for a detailed explanation: InsightSoftwareConsortium/ITK#343.

The solution in the above linked ITK issue is to change how HDF5 is linked by using `*_LIBRARY_NAMES` rather than `*_LIBRARIES`. An easier workaround that requires no changes in the DAGMC source itself is to set `HDF5_ROOT`, which causes `h5cc -show` to fail which then triggers a fallback in FindHDF5.cmake whereby it only searches for libhdf5.so and not all the transitive dependencies.

I'm not sure there's a good proper fix short of 1) the HDF5 feedstock on conda-forge using CMake rather than autoconf, or 2) FindHDF5.cmake being fixed to not hardcode paths to system libraries.

## Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] <s>Reset the build number to `0` (if the version changed)</s>
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by paulromano** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh [hunk 1]
```diff
@@ -32,6 +32,9 @@ cmake -DBUILD_MCNP5=OFF \
       -DBUILD_PIC=OFF \
       -DBUILD_RPATH=ON \
       -DDOUBLE_DOWN=OFF \
+      -DCMAKE_POLICY_DEFAULT_CMP0074=NEW \
+      -DHDF5_ROOT="${PREFIX}" \
+      -DHDF5_FIND_DEBUG=ON \
```
Comment (gonuke) [inline recipe/build.sh:37]: Does this have a continuing purpose? Or was it transient?

**Review by gonuke — CHANGES_REQUESTED** [review]
Do we need to leave the HDF5 find debug on?

File: recipe/build.sh (hunk 1 above)
Comment (paulromano) [reply recipe/build.sh:37]: Just debugging, although the fix I've proposed here didn't work as intended so I still have some digging to do :disappointed:

**Comment by paulromano** [comment]
Well, unfortunately my attempts to get around the HDF5 library linking issue appear to be unsuccessful. I'm going to submit a PR to DAGMC proper that can be incorporated in a future release.

**Pull request closed without merging**